File: tmp_fix_json.js
const fs = require('fs');
const path = require('path');

function fix(filePath) {
    let content = fs.readFileSync(filePath, 'utf8');

    // Fix the specific comma issue
    content = content.replace(/}\s*"fixNames"/g, '},\n        "fixNames"');

    try {
        // This will automatically deduplicate keys (last one wins)
        const data = JSON.parse(content);
        fs.writeFileSync(filePath, JSON.stringify(data, null, 4), 'utf8');
        console.log(`Successfully fixed and deduplicated ${filePath}`);
    } catch (e) {
        console.error(`Error parsing ${filePath}:`, e.message);
        // If it still fails, let's try to find where it's failing
        const lines = content.split('\n');
        const match = e.message.match(/at position (\d+)/);
        if (match) {
            const pos = parseInt(match[1]);
            console.error('Error at around:', content.substring(Math.max(0, pos - 50), Math.min(content.length, pos + 50)));
        }
    }
}

fix(process.argv[2]);

File: tmp_fix_json.py
import json
import sys

def fix_json(file_path):
    with open(file_path, 'r', encoding='utf-8') as f:
        content = f.read()
    
    # Try to parse it. If it fails, it might be due to the missing comma.
    try:
        data = json.loads(content)
        print(f"{file_path} is valid JSON")
    except json.JSONDecodeError as e:
        print(f"{file_path} is invalid: {e}")
        # Very specific fix for the commas I might have missed
        # Replace '}\n        "fixNames"' with '},\n        "fixNames"'
        content = content.replace('}\n        "fixNames"', '},\n        "fixNames"')
        try:
            data = json.loads(content)
            print(f"Fixed {file_path}")
            with open(file_path, 'w', encoding='utf-8') as f:
                json.dump(data, f, indent=4, ensure_ascii=False)
        except Exception as e2:
            print(f"Failed to fix {file_path}: {e2}")

if __name__ == "__main__":
    fix_json(sys.argv[1])
